packages/es-client/indexer.js

```diff
@@ -315,6 +317,30 @@ async function upsertGranule({
 
   const upsertDoc = updates;
 
+  let inline = 'ctx._source.putAll(params.doc)';
+  if (writeConstraints === true) {
+    // Because both API write and message write chains use the granule model to store records, in
+    // cases where createdAt does not exist on the granule, we assume overwrite protections are
+    // undesired behavior via business logic on the message write logic
+    inline = `
+    if ((ctx._source.createdAt === null || params.doc.createdAt >= ctx._source.createdAt)
+      && (params.doc.status != 'running' || (params.doc.status == 'running' && params.doc.execution != ctx._source.execution))) {
+      ctx._source.putAll(params.doc);
+    } else {
+      ctx.op = 'none';
+    }
+    `;
+    if (!updates.createdAt) {
+      inline = `
+        if (params.doc.status != 'running' || (params.doc.status == 'running' && params.doc.execution != ctx._source.execution)) {
```

Comment:
Note: this *does* have a logic flaw re: queued granule writes and needs to be addressed in a future ticket coming out of the 10/19 arch meeting.

Reply:
@Jkovarik was the ticket made?

Reply:
Yes: https://bugs.earthdata.nasa.gov/browse/CUMULUS-3117